tests/functional/history/test_db.py

```diff
@@ -0,0 +1,199 @@
+        writer_b = DatabaseRecordWriter(self.connection)
+        self._write_sequence_of_records(writer_a, [
+            {
+                'source': 'TEST',
+                'event_type': 'foo',
+                'payload': '',
+                'timestamp': 3
+            },
+            {
+                'source': 'TEST',
+                'event_type': 'bar',
+                'payload': '',
+                'timestamp': 1
+            },
+            {
+                'source': 'TEST',
+                'event_type': 'baz',
+                'payload': '',
+                'timestamp': 4
+            }
+        ])
+        self._write_sequence_of_records(writer_b, [
+            {
+                'source': 'TEST',
+                'event_type': 'qux',
+                'payload': '',
+                'timestamp': 2
+            },
+            {
+                'source': 'TEST',
+                'event_type': 'zip',
+                'payload': '',
+                'timestamp': 6
+            }
+        ])
+        reader = DatabaseRecordReader(self.connection)
+        cursor = self._get_cursor()
+        cursor.execute(
+            'select id from records where event_type = "foo" limit 1')
+        identifier = cursor.fetchone()['id']
+
+        # This should select only the three records from writer_a since we
+        # are explicitly looking for the records that match the id of the
+        # foo event record.
+        records = [record for record in reader.yield_records(identifier)]
+        self.assertEqual(len(records), 3)
+        for record in records:
+            record_id = record['id']
+            self.assertEqual(record_id, identifier)
+
+    def test_can_read_most_recent_records(self):
+        writer_a = DatabaseRecordWriter(self.connection)
+        writer_b = DatabaseRecordWriter(self.connection)
+        self._write_sequence_of_records(writer_a, [
+            {
+                'source': 'TEST',
+                'event_type': 'foo',
+                'payload': '',
+                'timestamp': 3
+            },
+            {
+                'source': 'TEST',
+                'event_type': 'bar',
+                'payload': '',
+                'timestamp': 1
+            }
+        ])
+        self._write_sequence_of_records(writer_b, [
+            {
+                'source': 'TEST',
+                'event_type': 'baz',
+                'payload': '',
+                'timestamp': 2
+            }
+        ])
+
+        # Since the foo and bar events were written by the writer_a they all
+        # share an id. foo was written at time 3 which makes it the most
+        # recent, so when we call get_latest_records we should get the
+        # foo and bar records only.
+        reader = DatabaseRecordReader(self.connection)
+        records = set([record['event_type'] for record
+                       in reader.yield_latest_records()])
+        self.assertEqual(set(['foo', 'bar']), records)
```

Comment:
Should we add tests to make sure the `yield_latest_records` and `yield_records` methods return items by timestamps?